play_motion:
  motions:
    pregrasp_torso:
      joints: [torso_lift_joint, arm_1_joint,
      arm_2_joint, arm_3_joint, arm_4_joint, arm_5_joint,
      arm_6_joint, arm_7_joint]
      points:
      - positions: [0.34, 0.15, -1.3, -0.2, 1.92, -1.57, 1.37, 0.0]
        time_from_start: 1.0
      - positions: [0.34, 0.15, 0.1, 0.3, 1.92, -1.57, 1.37, 0.0]
        time_from_start: 2.0
      - positions: [0.34, 0.15, 0.1, 0.3, 0.1, -1.57, 0.2, 0.0]
        time_from_start: 3.0
      - positions: [0.34, 0.15, 0.1, 0.3, 0.1, -1.57, 0.2, 0.0]
        time_from_start: 4.0
      - positions: [0.34, 0.15, 0.1, -3.2, 0.1, -1.57, 0.2, 0.0]
        time_from_start: 6.5
      - positions: [0.15, 0.15, 0.1, -3.2, 1.5, -1.57, 0.2, 0.0]
        time_from_start: 8

    pregrasp:
      joints: [arm_1_joint, arm_2_joint, arm_3_joint, arm_4_joint, arm_5_joint, arm_6_joint, arm_7_joint]
      points:
      - positions: [0.15, -1.3, -0.27, 1.91, -1.57, 1.37, 1.3]
        time_from_start: 1.0
      - positions: [0.15, -1.3, -0.27, 2.28, -1.57, 0.97, 1.3]
        time_from_start: 2.5
      - positions: [0.15, 0.3, -0.27, 2.28, -1.57, 1, 1.3]
        time_from_start: 4.0
      - positions: [0.15, 0.3, -0.27, 0.2, -1.57, 1.3, 1.3]
        time_from_start: 6.5
      - positions: [0.15, 0.3, -0.27, 0.2, -1.57, 0.0, 1.3]
        time_from_start: 8
      - positions: [0.15, 0.3, -3, 0.2, -1.57, 0.0, 0.0]
        time_from_start: 10
      - positions: [0.15, 0.3, -3, 1.5, 0, 0.0, 0.0]
        time_from_start: 12
    
    preplace:
      joints: [arm_1_joint, arm_2_joint, arm_3_joint, arm_4_joint, arm_5_joint, arm_6_joint, arm_7_joint]
      points:
      - positions: [0.15, 0.3, -3, 1.5, 0, 0.0, 0.0]
        time_from_start: 4

    move_safe:
      joints: [torso_lift_joint, arm_1_joint,
      arm_2_joint, arm_3_joint, arm_4_joint, arm_5_joint,
      arm_6_joint, arm_7_joint]
      points:  
      - positions: [0.34, 0.10, 0.47, -0.20, 1.56, 1.60, 0.25, 1.19]
        time_from_start: 10.5

    open_gripper:
      joints: [gripper_right_finger_link, gripper_left_finger_link]
      points:  
      - positions: [0.44, 0.44]
        time_from_start: 1.0
